state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -42,84 +44,116 @@ pub const PALLET_VERSION: StorageVersion = StorageVersion::new(3);
 
 pub const SWAP_DELAY_BLOCKS: u32 = 2;
 
-#[derive(Clone, Debug, PartialEq, Eq, Encode, Decode, TypeInfo, MaxEncodedLen)]
-pub struct Swap {
-	swap_id: SwapId,
-	pub from: Asset,
-	pub to: Asset,
-	input_amount: AssetAmount,
-	swap_type: SwapType,
+struct SwapCtx {
```

Comment:
nit: I feel like the naming of Swap/SwapCtx is a bit confusing. 

To me, 'Ctx' implies (immutable) context, ie. 'what is the context of this swap', which could be 'ccm swap' or 'network fee swap'. 

So maybe the names should be reversed, or instead of SwapCtx we could name it `SwapProgress` or `SwapState` or something that makes it clear that this a container for mutable state and can be updated.

Reply:
I'm not sure. To me "context" does not necessarily imply "immutable context". To me "context" sounds like it may contain the swap itself (which it does), whereas SwapProgress sound more like a single value, e.g. InProgress/Done.  `SwapState` is probably OK too, though I don't see a big different tbh.

Reply:
fwiw `State` sounds more intuitive to me for this data

Reply:
OK, renaming to `SwapState`